app/services/bolognese/writers/hyku_addons_work_form_fields_writer.rb

```diff
@@ -498,13 +500,11 @@ def ensure_creator_from_editor!
         end
 
         def process_editor_contributors!
-          editors = Proc.new { |item| item["contributor_contributor_type"] == "Editor" }
+          editors = proc { |item| item["contributor_contributor_type"] == "Editor" }
 
           # If we do not have editors, they might be missing from the fields for this work type.
           # This is so that we can reliably use them later on in the callbacks chain
-          if @form_data["editor"].blank? && @form_data["contributor"].any?(&editors)
-            @form_data["editor"] = @form_data["contributor"].select(&editors)
-          end
+          @form_data["editor"] = @form_data["contributor"].select(&editors) if @form_data["editor"].blank? && @form_data["contributor"].any?(&editors)
```

Comment:
Another sanity check. An empty array of `editors` would pass the second check of the if expression. Is that what you want?